lib/simple-cookie-manager.js

```diff
@@ -4,24 +4,50 @@
  * Released under the GNU GENERAL PUBLIC LICENSE
 */
 
-var SimpleCookieManager = function(){
-  this.readCookie = function(name){
- var _search = name + "=";
- if (document.cookie.length > 0) {  
-  _cookiePos = document.cookie.indexOf(_search);
-  if (_cookiePos != -1) { 
-   _cookiePos += _search.length;  
-   _finish = document.cookie.indexOf(";", _cookiePos); 
-   if (_finish == -1) 
-     _finish = document.cookie.length;
-   return unescape(document.cookie.substring(_cookiePos, _finish));
-  }
- }
-}
- this.writeCookie = function(name, val, h){
-  var _today = new Date();
-  var _cookend = new Date();
-  _cookend.setTime(_today.getTime() + 1000*60*60*h);
-  document.cookie = name + "=" + escape(val) + "; expires=" + _cookend.toGMTString();
-}
-}
\ No newline at end of file
+var SimpleCookieManager = function(){};
+
+/**
+ *
+ * @param name
+ * @returns {*}
+ */
+SimpleCookieManager.prototype.readCookie = function(name) {
+    var search = name + "=";
+
+    if (document.cookie.length <= 0) {return "";}
+
+    var cookiePos = document.cookie.indexOf(search);
+
+    if (cookiePos === -1) {return "";}
+
+    cookiePos += search.length;
+
+    var finish = document.cookie.indexOf(";", cookiePos);
+
+    if (finish === -1) {
+        finish = document.cookie.length;
+    }
+
+    return decodeURIComponent(document.cookie.substring(cookiePos, finish));
+};
+
+/**
+ *
+ * @param name
+ * @param value
+ * @param hours
+ */
+SimpleCookieManager.prototype.writeCookie = function(name, value, hours) {
+    var today = new Date();
+    var cookEnd = new Date();
+    var timeInMilliseconds = 1000 * 60 * 60 * hours;
+
+    cookEnd.setTime(today.getTime() + timeInMilliseconds);
+
+    var safeValue =  encodeURIComponent(value)
+        .replace(/%(23|24|26|2B|3A|3C|3E|3D|2F|3F|40|5B|5D|5E|60|7B|7D|7C)/g, decodeURIComponent);
+    var safeKey = encodeURIComponent(name)
+        .replace(/%(23|24|26|2B|5E|60|7C)/g, decodeURIComponent);
+
+    document.cookie = [safeKey, "=", safeValue, "; expires=", cookEnd.toGMTString()].join('');

```

Comment:
this is arguably faster than concatanating strings with `+`; though it's a microoptimization, and it does not matter anymore in modern user agents anyway.